src/main/java/org/pesmypetcare/webservice/controller/petmanager/KcalAverageRestController.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    /**
+     * Get one kcalAverage of the pet with the specified date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param petDate Date of the kcalAverage instance
+     * @return The KcalAverageEntity with the data of kcalAverage
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/{petDate}")
+    public KcalAverageEntity getKcalAverageByDate(@PathVariable String owner, @PathVariable String petName,
+                                        @PathVariable String petDate) throws DatabaseAccessException {
+        return kcalAverageService.getKcalAverageByDate(owner, petName, petDate);
+    }
+
+    /**
+     * Gets all the specified kcalAverages of one pet.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @return The list with all of the kcalAverages
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}")
+    public List<Map<String, Object>> getAllKcalAverage(@PathVariable String owner, @PathVariable String petName)
+        throws DatabaseAccessException {
+        return kcalAverageService.getAllKcalAverage(owner, petName);
+    }
+
+    /**
+     * Gets the data from all the meals eaten by the pet between the initial and final date not including them.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param initialDate Initial Date
+     * @param finalDate Final Date
+     * @return The List containing all the kcalAverages in the specified time
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/between/{initialDate}/{finalDate}")
+    public List<Map<String, Object>> getAllKcalAveragesBetween(@PathVariable String owner, @PathVariable String petName,
+                                                          @PathVariable String initialDate,
+                                                          @PathVariable String finalDate)
+        throws DatabaseAccessException {
+        return kcalAverageService.getAllKcalAveragesBetween(owner, petName, initialDate, finalDate);
+    }
+
+    /**
+     * Updates one of the kcalAverages specified with the date and the pet name.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param petDate Date of the kcalAverage instance
+     * @param valueMap Entity that contains the value that the field will have. The new field value needs to have the
+     *                 key "value"
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @PutMapping("/{owner}/{petName}/{petDate}")
+    public void updateKcalAverage(@PathVariable String owner, @PathVariable String petName, @PathVariable String petDate,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 121).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)